notebook_index.yml

```diff
@@ -115,6 +115,8 @@ contents:
               - pump: Pump unit model with iapws property package
               - heat exchanger 0D: Heat Exchanger 0D unit model heating a benzene-toluene mixture using steam
               - skeleton unit: Skeleton unit model demonstrating custom variables, ports, and constraints
+              - statejunction: Statejunction unit model with modular property package
```

Comment:
I cannot comment on the icons themselves, so I will mention it here: I do not think a box is the best icon for a state junction. It is essentially just a single node (StateBlock) and not an actual unit, so a filled circle might be a better choice.

Reply:
Also, do we need to have multiple copies of the icon in different places? It would be nice if we could have a single copy in a central locations (e.g. the `common` folder).

Reply:
For the multiple copies of icons, I agree that we should just have them all in one place. Perhaps this should be part of a separate PR tho since it would involve moving around a bunch of files. So it might just make more sense to make this change after migrating everything over to the new examples repo.